shopfloor_reception/services/reception.py

```diff
@@ -302,91 +322,165 @@ def _scan_document__by_picking(self, barcode):
+        if self.work.menu.allow_return:
+            response = self._scan_line__by_product__return(picking, product)
+            if response:
+                return response
+        # Otherwise, the picking isn't a return, and should be a regular reception
+        move = picking.move_lines.filtered(lambda m: m.product_id == product)
+        message = self._check_move_available(move, "product")
+        if message:
+            return self._response_for_select_move(
+                picking,
+                message=message,
+            )
+        return self._scan_line__find_or_create_line(picking, move)
 
-    def _scan_line__by_lot(self, picking, barcode):
-        line = picking.move_line_ids.filtered(
-            lambda l: barcode == l.lot_id.name or barcode == l.lot_name
-        )
-        move = line.move_id
+    def _scan_line__by_packaging__return(self, picking, packaging):
         search = self._actions_for("search")
-        lot = search.lot_from_scan(barcode)
-        if not move:
-            line = picking.move_line_ids.filtered(
-                lambda l: not l.lot_id
-                and not l.lot_name
-                and l.product_id == lot.product_id
+        origin_moves = search.origin_move_from_scan(picking.origin)
+        origin_moves.filtered(lambda m: m.picking_id.picking_type_code == "outgoing")
+        origin_moves_for_packaging = origin_moves.filtered(
+            lambda m: packaging in m.product_id.packaging_ids
+        )
+        if origin_moves and not origin_moves_for_packaging:
+            message = self.msg_store.product_not_found_in_picking()
+            return self._response_for_select_move(picking, message=message)
+        # If we have an origin move, create the return move, and go to next screen
+        if origin_moves_for_packaging:
+            return_move = self._scan_line__create_return_move(
+                picking, origin_moves_for_packaging
             )
-            if line:
-                return self._select_line(picking, line, move)
-        else:
-            message = self._check_move_available(move, message_code="lot")
-            if message:
-                return self._response_for_select_move(
-                    picking,
-                    message=message,
-                )
-            if lot:
-                return self._select_line_from_lot(picking, move, barcode)
+            return_move._action_confirm()
+            return self._scan_line__find_or_create_line(
+                picking, return_move, packaging.qty
+            )
+
+    def _scan_line__by_packaging(self, picking, packaging):
+        if self.work.menu.allow_return:
+            response = self._scan_line__by_packaging__return(picking, packaging)
+            if response:
+                return response
+        move = picking.move_lines.filtered(
+            lambda m: packaging in m.product_id.packaging_ids
+        )
+        message = self._check_move_available(move, "packaging")
```

Comment:
Same here.
A move may already exist and should be used before creating a new return move. If you create a return in odoo backend, you need to be able to process the move even is the option is enabled. Otherwise, the move you create can get merged with the existing one and it will crash.